foundry/game/gfx/GraphicsSet.py

```diff
@@ -119,6 +124,23 @@ def _read_in(self, segments):
         for segment in segments:
             self._read_in_chr_rom_segment(segment, self._data)
 
+    def _heuristic_bg_pages(self, bg_page_bytes: bytes, fallback_addr: int) -> int:
+        """Searches through the ROM for the main array responsible for rendering the correct graphics.
+        Currently the heuristics in order of precedence are:
+
+        1. Search for the bytes from the stock ROM (given as the bg_page_bytes argument)
+            - When changing up the assembly, it's probably rare that this array changes, but it can
+              easily change position in the bank, so this is most likely to be correct if it is found.
+
+        2. Use the given `fallback_addr`.
+
+        TODO: Do more/better heuristics than searching the stock bytes.
+        """
+        bgpages_addr = ROM().search(bg_page_bytes)
```

Comment:
I'd add the start of the PRG to search for it and maybe a check, that the returned address is within the expected range.

Reply:
Resolved by f976918
The range check is inherent in the `search_bank` now.